### DISCUSSION TITLE
Default value for measurable habits

### DISCUSSION THREAD

**Discussion by mitridatik** [iSoron/uhabits#1974] [Feature Requests]
Often, you need to stick to some value in measurable habits, and in most cases, it is the same value, like 30 mins, for example. Optional default value for measurable habit will do the trick to save time on data entrance.

**Comment by yan32323** [comment]
I'd like to add to that, right now I use it to track the time I use to track the number of 5 minute courses I do before I go to bed, setting my goal to at least 6. Opening the input menu to only add 1 to the counter is getting old sometimes.

Right now, with measurable habits, both the long press and short press have the same action : opening the input menu.

For the first step, we could set the short press to only add 1 to the counter, and the long press to show the menu to enter a number. Of course, it would be opt-in with an option under interface in the options menu (like the "validate habits with a short press" option we have right now)

Going further, we could set a personalised amount to add on short press. This would be on a "per habit" customisation.

Going further we could set multiple action triggers on the same "per habit" customisation.
Example : 
short press : +1
double short press : +2
triple short press : +10

To keep the UI simple and elegant, of course, this menu could be hidden behind a button (like the color selector, but besides the target section) and/or completely hidden until the user opts-in to this feature in the app's settings.

Thank you.

**Comment by ezio-melotti** [comment]
I would also like this feature.  I have several measurable habit that I do every day.  For each of them, every day I have to tap on each of them, enter the numeric goal (always the same), and press save:
![Screenshot_20250709_051641_Habits](https://github.com/user-attachments/assets/0bfd1bd3-04d9-4247-976e-5ed26b15f4b6)

I suggest the following (simplest) implementation (in bold the suggested change):
* tap on the habit I want to update
* the popup shown in the screenshot appears:
  * **if the current value is `?` (no value entered yet): show the default target (e.g. 30, instead of 0)**
  * if the current value is a number: show the current number (current behavior)
* tap `SAVE` to enter the value

This should be doable by simply adding a condition here (I think):
https://github.com/iSoron/uhabits/blob/a9acbd6cabd6da33d2f0b8363f9651d7e75c2d37/uhabits-android/src/main/java/org/isoron/uhabits/activities/common/dialogs/NumberDialog.kt#L55-L60

This simple change will halve the taps I need for each habit and prevent typos.

More complex/customizable implementations are also possible, but I would start with this and possibly iterate on it later.
